File: web/src/types/social.ts
export interface Friend {
    user_id: number;
    username: string;
    display_name: string;
    online: boolean;
    last_seen: string | null;
    close_friend: boolean;
    group_ids: number[];
    guild: GuildTag | null;
}

export interface GuildTag {
    id: number;
    tag: string;
    color: string;
    icon: string;
}

export interface FriendGroup {
    id: number;
    name: string;
}

export interface FriendRequestItem {
    id: number;
    user_id: number;
    username: string;
    display_name: string;
    created_at: string;
}

export interface FriendRequestsResponse {
    incoming: FriendRequestItem[];
    outgoing: FriendRequestItem[];
}

export interface TradeRequestItem {
    id: number;
    user_id: number;
    username: string;
    display_name: string;
    created_at: string;
}

export interface TradeRequestsResponse {
    incoming: TradeRequestItem[];
    outgoing: TradeRequestItem[];
}

export interface TradePulse {
    active_session_id: number | null;
    incoming_unseen: TradeRequestItem[];
    incoming_ids: number[];
    outgoing_ids: number[];
}

File: backend/app/schemas/social.py
"""
Schemas — amis, demandes d'ami, demandes d'échange (placeholder).
"""

from datetime import datetime
from pydantic import BaseModel, Field


class FriendOut(BaseModel):
    user_id: int
    username: str
    display_name: str
    online: bool
    last_seen: datetime | None = None
    close_friend: bool = False
    group_ids: list[int] = []


class FriendGroupOut(BaseModel):
    id: int
    name: str


class FriendGroupBody(BaseModel):
    name: str = Field(min_length=1, max_length=30)


class SendFriendRequestBody(BaseModel):
    username: str = Field(min_length=1, max_length=20)


class FriendRequestOut(BaseModel):
    id: int
    user_id: int  # l'AUTRE joueur impliqué dans la demande
    username: str
    display_name: str
    created_at: datetime


class FriendRequestsResponse(BaseModel):
    incoming: list[FriendRequestOut] = []
    outgoing: list[FriendRequestOut] = []


class TradeRequestOut(BaseModel):
    id: int
    user_id: int  # l'AUTRE joueur impliqué
    username: str
    display_name: str
    created_at: datetime


class TradeRequestsResponse(BaseModel):
    incoming: list[TradeRequestOut] = []
    outgoing: list[TradeRequestOut] = []


class TradePulseOut(BaseModel):
    """État condensé interrogé en continu par le client (échange lancé,
    nouvelles demandes à afficher, listes à rafraîchir)."""
    active_session_id: int | None = None
    active_other_display_name: str | None = None
    incoming_unseen: list[TradeRequestOut] = []
    incoming_ids: list[int] = []
    outgoing_ids: list[int] = []


class SendTradeRequestBody(BaseModel):
    username: str = Field(min_length=1, max_length=20)
